File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int getNum(int n)
{
    int cnt = 0;
    for (int i = 1; i * i <= n; i++)
    {
        if (n % i == 0)
        {
            cnt++;
            if (i * i < n)
                cnt++;
        }
    }
    return cnt;
}

int solution(int number, int limit, int power)
{
    int answer = 0;
    for (int i = 1; i <= number; i++)
    {
        int result = getNum(i);
        if (result > limit)
            answer += power;
        else
            answer += result;
    }
    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    
    public static int go(int number) {
        if (number <= 2) return number;
        
        int count = 2;
        for (int i = 2; i <= number / 2; i++) {
            if (number % i == 0) count++;
        }
        
        return count;
    }
    
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        for (int i = 1; i <= number; i++) {
            int val = go(i);
            if (val > limit) {
                answer += power;
            } else {
                answer += val;
            }
        }
        
        return answer;
    }
}